crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -544,6 +561,26 @@ impl ShaderProcessor {
                 let current_valid = scopes.last().unwrap().is_accepting_lines();
 
                 scopes.push(Scope::new(current_valid && new_scope));
+            } else if let Some(cap) = self.define_regex.captures(line) {
+                let def = cap.get(1).unwrap();
+                let name = def.as_str().to_string();
+
+                if let Some(val) = cap.get(2) {
+                    if let Ok(val) = val.as_str().parse() {
+                        shader_defs_unique.insert(name.clone(), ShaderDefVal::UInt(name, val));
+                    } else if let Ok(val) = val.as_str().parse() {
+                        shader_defs_unique.insert(name.clone(), ShaderDefVal::Int(name, val));
+                    } else if let Ok(val) = val.as_str().parse() {
+                        shader_defs_unique.insert(name.clone(), ShaderDefVal::Bool(name, val));
```

Comment:
The type inference here makes for confusing-looking code. Could you use explicit types in the `.parse::<T>()` to make it clearer?

Reply:
Or... could we go C-preprocessor-style and just use string replacement? When I got down to the tests with `DEFUINT 1` and `DEFINT -1` it felt weird to define the type of the value here given that `1` in WGSL is signed and `1u` is unsigned.

Reply:
> The type inference here makes for confusing-looking code. Could you use explicit types in the `.parse::<T>()` to make it clearer?

but I like the type magic that makes the same code not have the same result 😄 

> Or... could we go C-preprocessor-style and just use string replacement? When I got down to the tests with `DEFUINT 1` and `DEFINT -1` it felt weird to define the type of the value here given that `1` in WGSL is signed and `1u` is unsigned.

keeping it as string all along would make harder to use the value in if condition